pkg/validation/internal/operatorhub.go

```diff
@@ -81,41 +101,202 @@ func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
+func validateHubDeprecatedAPIS(bundle *manifests.Bundle, k8sVersion string) ([]error, []error) {
 	var errs []error
 	var warns []error
+	k8s1_22, _ := semver.Make("1.22.0")
+	k8s1_16, _ := semver.Make("1.16.0")
+	hasK8sVersion := len(k8sVersion) > 0
 
-	if csv.Spec.Provider.Name == "" {
-		errs = append(errs, fmt.Errorf("csv.Spec.Provider.Name not specified"))
+	// Transform the key/option k8sVersion in semver Version to compare
+	var semverK8sVersion semver.Version
+	if hasK8sVersion {
+		version := k8sVersion
+		if len(strings.Split(k8sVersion, ".")) == 2 {
+			// if the value found is like 1.16 then, we concat with .0 to check it with semver
+			version += ".0"
+		}
+		var err error
+		semverK8sVersion, err = semver.Make(version)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("invalid value informed via the k8s key option : %s", k8sVersion))
+		} else {
+			// we might want to return it as info instead of warning in the future.
+			warns = append(warns, fmt.Errorf("checking APIs against Kubernetes version : %s", k8sVersion))
+		}
 	}
 
-	for _, maintainer := range csv.Spec.Maintainers {
-		if maintainer.Name == "" || maintainer.Email == "" {
-			errs = append(errs, fmt.Errorf("csv.Spec.Maintainers elements should contain both name and email"))
+	// Transform the spec minKubeVersion in semver Version to compare
+	var semverMinKube semver.Version
+	if len(bundle.CSV.Spec.MinKubeVersion) > 0 {
+		minKube := bundle.CSV.Spec.MinKubeVersion
+		if len(strings.Split(bundle.CSV.Spec.MinKubeVersion, ".")) == 2 {
+			minKube += ".0"
 		}
-		if maintainer.Email != "" {
-			_, err := mail.ParseAddress(maintainer.Email)
-			if err != nil {
-				errs = append(errs, fmt.Errorf("csv.Spec.Maintainers email %s is invalid: %v", maintainer.Email, err))
+		var err error
+		if semverMinKube, err = semver.Make(minKube); err != nil {
+			errs = append(errs, fmt.Errorf("unable to use csv.Spec.MinKubeVersion to verify the CRD/Webhook apis "+
+				"because it has an invalid value: %s", bundle.CSV.Spec.MinKubeVersion))
+		}
+	}
+
+	// If the k8sVersion was informed then, the check is against this version (it means that it is to verify
+	// the operator bundle against the k8s version which it is intended to be distributed).
+	// However, if the minKubeVersion not be informed then, it means that the operator can be distributed in any version.
+	// To ensure this scenario, when the k8sVersion is not informed, the check is made and a warning is raised
+	// if the deprecated/unsupported APIs are found. See that the error only can be safely raised when the
+	// minKubeVersion value is >=1.22.
+	if !hasK8sVersion || semverK8sVersion.GE(k8s1_16) {
+		isUnsupported := semverK8sVersion.GE(k8s1_22) || semverMinKube.GE(k8s1_22)
+
+		// Note that if the operator bundle has both versions then, it will only raise a warning since it might
+		// can means that the operator has been covering both scenarios.
+		if hasDeprecatedAPIs(bundle) {
```

Comment:
I would make these checks and resulting error messages specific to the API type. Getting a list of possible GKs is confusing, because one or multiple of them could exist in a project.

Reply:
done